### ISSUE TITLE
100% CPU usage used with podman

### ISSUE THREAD

**Issue by supermar1010** [rootless-containers/slirp4netns#157]
Hi,
I'm using this project with podman. I'm not sure if this is related to podman or to slirp4netns. So I'm posting it here and at podman. 
I have one container with networking mode host, all the others have no setting so I guess podman namespaces them I have four containers running, one with root permissions.
This is the command using all 100% of the CPU on one core.
`/usr/bin/slirp4netns --api-socket /run/user/1000/libpod/tmp/18c14ae7aed050b0063c4b5f75af384721a280011fbf95a7e85042112af7458d.net --disable-host-loopback --mtu 65520 -c -e 3 -r 4 --netns-type=path /run/user/1000/netns/cni-7913c5a5-8763-5636-16ad-286fa2a0a376 tap0`

As far I can tell everything works as expected. So no problems on the functional site. 
Any idea what is causing this?

**Comment by AkihiroSuda** [comment]
With what workloads?

**Comment by supermar1010** [comment]
No load at all. I guess a maximum of 1 site call a second.
What I forgot, there is a tor middle relay running which relays about 1MBit/s.
So no significant load I guess. 
The CPU load dropped to normal levels after approximately half an hour. So it seems to be kind of sporadic?

**Comment by supermar1010** [comment]
Another thing I just noticed, it seems to limit my tor middle relay as I'm only having a throughput of 1MBit/s , but my network connection is capable of 200MBit/s so it seems it limits the throughput a lot.

**Comment by AkihiroSuda** [comment]
https://github.com/rootless-containers/slirp4netns/blob/master/slirp4netns.1.md#bugs

this might be related

**Comment by AkihiroSuda** [comment]
changing `slirp4netns --mtu` to 1500 may change the situation, though I'm not sure podman supports changing mtu

**Comment by rhatdan** [comment]
@giuseppe @mheon @baude is this something Podman should support?  Would it make sense for rootfull containers as well?  Interaction with CNI?

**Comment by giuseppe** [comment]
@AkihiroSuda I remember we already discussed it but cannot find the comment now.  Isn't #128 something that can be fixed with changing the receive buffer for the slirp4netns socket only instead of changing the default for the network namespace?

**Comment by AkihiroSuda** [comment]
Is this still an issue?

**Comment by AkihiroSuda** [comment]
Closing, but feel free to reopen if this is still an issue

**Comment by casperfrx** [comment]
I know this is going to be a complete necro of an issue in 2025 but I of course Googled first and I figured it would still be relevant to post the issue where it has context. Please do tell me if I should open a new ticket and/or if I should approach it differently.

I'm having the same issue at this moment and I'm basically downloading a +- 6GB Windows ISO image from a Caddy server running in my rootless Caddy container. When that download is happening, any other requests to the Caddy server and fileserver listing seems to break and/or timeout. The download however continues but slirp4netns seems to run at 100%. Once the download is done and that single core peak is gone, everything works again as it should be. The Caddy server runs as a rootless Caddy container and works direclty with its Caddy native fileserver as the sole program to serve the directory listing/browsing.

To be 100% sure it's the download I tried using multiple browsers on different devices and they all timeout. I already tried to limit CPU usage with Docker CPU limits and niceness but no dice. Any ideas?

Oh before I forget to mention: this is a rootless Docker container, not a Podman deployment.

![Image](https://github.com/user-attachments/assets/a20f51c0-1d7f-477f-b584-27f386562451)

**Comment by casperfrx** [comment]
I forgot to tag @AkihiroSuda, I apologize for necroing the issue, honestly.
